File: 01-Creational/1.1 Abstract Factory/csharp/BMWDoor.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AbstractFactory
{
    class BMWDoor:Door
    {
        public BMWDoor(int numberofParts) : base(numberofParts) { }

        public override void bringPiecesFromInventory()
        {
            Console.WriteLine("gathering all the pieces to manufacture: Door for BMW");
        }
    }
}

File: 01-Creational/1.1 Abstract Factory/c++/BMWDoor.cpp
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

/* 
 * File:   BMWDoor.cpp
 * Author: casertillo
 * 
 * Created on 30 January 2017, 20:43
 */

#include "BMWDoor.h"
#include "iostream"

BMWDoor::BMWDoor(int numberofParts):Door(numberofParts) {
}

void BMWDoor::bringPiecesFromInventory(){
    std::cout << "gathering all the pieces to manufacture: Door for BMW \n";
}


